src/mumble/CustomElements.cpp

```diff
@@ -180,43 +180,51 @@ void ChatbarTextEdit::insertFromMimeData(const QMimeData *source) {
 }
 
 bool ChatbarTextEdit::sendImagesFromMimeData(const QMimeData *source) {
-	if (source->hasImage()) {
-		// Process the image pasted onto the chatbar.
-		if (g.bAllowHTML) {
+	if (g.bAllowHTML) {
+		if (source->hasImage()) {
+			// Process the image pasted onto the chatbar.
 			QImage image = qvariant_cast< QImage >(source->imageData());
-
-			QString imgHtml = QLatin1String("<br />") + Log::imageToImg(image);
-
-			if ((g.uiImageLength == 0) || static_cast< unsigned int >(imgHtml.length()) < g.uiImageLength) {
-				emit pastedImage(imgHtml);
+			if (emitPastedImage(image)) {
 				return true;
 			} else {
 				g.l->log(Log::Information, tr("Unable to send image: too large."));
 			}
-		}
-	} else if (source->hasUrls()) {
-		// Process the files dropped onto the chatbar. URLs here should be understood as the URIs of files.
-		QList< QUrl > urlList = source->urls();
-
-		int count = 0;
-		for (int i = 0; i < urlList.size(); ++i) {
-			QString path = urlList[i].toLocalFile();
-			QImage image(path);
 
-			if (image.isNull())
-				continue;
-
-			QString imgHtml = QLatin1String("<br />") + Log::imageToImg(image);
-
-			if (static_cast< unsigned int >(imgHtml.length()) < g.uiImageLength) {
-				emit pastedImage(imgHtml);
-				++count;
-			} else {
-				g.l->log(Log::Information, tr("Unable to send image %1: too large.").arg(path));
+		} else if (source->hasUrls()) {
+			// Process the files dropped onto the chatbar. URLs here should be understood as the URIs of files.
+			QList< QUrl > urlList = source->urls();
+
+			int count = 0;
+			for (int i = 0; i < urlList.size(); ++i) {
+				QString path = urlList[i].toLocalFile();
+				QImage image(path);
+
+				if (image.isNull())
+					continue;
+				if (emitPastedImage(image)) {
+					++count;
+				} else {
+					g.l->log(Log::Information, tr("Unable to send image %1: too large.").arg(path));
+				}
 			}
+
+			return (count > 0);
 		}
+	}
+	g.l->log(Log::Information, tr("This server does not allow sending images."));
+	return false;
+}
 
-		return (count > 0);
+bool ChatbarTextEdit::emitPastedImage(QImage image) {
+	int maxSize = 65536;
```

Comment:
What is this number?

Reply:
It kind of is [this number](https://github.com/mumble-voip/mumble/blob/master/src/mumble/Log.cpp#L542). So the old behaviour of the client was to check whether the image is larger than imagemessagelength. If it was it was rejected. Otherwise it was processed but still compressed to 64 kB. So in a sense the size check in the beginning was superfluous. It's hard to explain in words; it's easier to see in [the code](https://github.com/mumble-voip/mumble/blob/master/src/mumble/CustomElements.cpp#L182).

So I thought there must have been a reason for that so I thought we better compress the image to 64 kB or imagemessagelength, whatever is smaller. But maybe we can just drop the 64 kB. I have also found some reference to it [here](https://wiki.mumble.info/wiki/Murmur.ini#imagemessagelength).

Reply:
I just checked the git history to see if at some point there was a reasoning for this specific value, but there was none given.

The 64kB size limit seems to apply to clients v1.2.3 or older. I don't think we have to take this into account anymore :thinking:

As I don't see a reason why we should have this boundary in there. Thus I'd vote for dropping it :point_up:

Reply:
Thanks for looking into the history. Ok let's drop it.